scripts/eurostat/health_determinants/common/replacement_functions.py

```diff
@@ -17,357 +17,210 @@
-            'TOTAL': 'Total',
-            'NSM': 'NonSmoker',
-            'SM_CUR': 'Smoking',
-            'SM_DAY': 'Smoking_Daily',
-            'SM_OCC': 'Smoking_Occasional',
-            'SM_LT20D': 'LessThan20CigarettesPerDay',
-            'SM_GE20D': '20OrMoreCigarettesPerDay',
-            'DSM_GE20': '20OrMoreCigarettesPerDay',
-            'DSM_LT20': 'LessThan20CigarettesPerDay'
-        }
-    })
-    return df
-
-
-def _replace_smoking_frequenc(df: pd.DataFrame) -> pd.DataFrame:
-    """
-    Replaces values of a single column into true values
-    from metadata returns the DF
-
-    Args: df (pd.DataFrame): df as the input, to change column values
-
-    Returns: df (pd.DataFrame): modified df as output
-    """
-    df = df.replace({
-        'frequenc': {
-            'DAY_GE1HD': 'AtLeastOneHourPerDay',
-            'DAY_LT1HD': 'LessThanOneHourPerDay',
-            'LT1W': 'LessThanOnceAWeek',
-            'GE1W': 'AtLeastOnceAWeek',
-            'RAR_NVR': 'RarelyOrNever',
-            'DAY': 'Smoking_Daily',
-            'FMR': 'FormerSmoker_Formerly',
-            'OCC': 'Smoking_Occasional',
-            'NVR': 'Smoking_NeverUsed'
-        }
-    })
-    return df
-
-
-def _replace_lev_perc(df: pd.DataFrame) -> pd.DataFrame:
-    """
-    Replaces values of a single column into true values
-    from metadata returns the DF.
-
-    Args: df (pd.DataFrame): df as the input, to change column values
-
-    Returns: df (pd.DataFrame): modified df as output
-    """
-    df = df.replace(
-        {'lev_perc': {
-            'STR': 'Strong',
-            'INT': 'Intermediate',
-            'POOR': 'Poor'
-        }})
-    return df
-
-
-def _replace_assist(df: pd.DataFrame) -> pd.DataFrame:
-    """
-    Replaces values of a single column into true values
-    from metadata returns the DF.
-
-    Args: df (pd.DataFrame): df as the input, to change column values
-
-    Returns: df (pd.DataFrame): modified df as output
-    """
-    df = df.replace({
-        'assist': {
-            'PROV': 'ProvidingInformalCare',
-            'PROV_R': 'Relatives_ProvidingInformalCare',
-            'PROV_NR': 'NonRelatives_ProvidingInformalCare',
-            'NPROV': 'NotProvidingInformalCare'
-        }
-    })
-    return df
+    Returns:
+        (pd.DataFrame): Updated DataFrame
+    """
+    mapper = {
+        'sex': _SEX,
+        'physact': _PHYSACT,
+        'isced11': _ISCED11,
+        'quant_inc': _QUANT_INC,
+        'frequenc': _FREQUENC,
+        'deg_urb': _DEG_URB,
+        'levels': _LEVELS,
+        'duration': _DURATION,
+        'c_birth': _C_BIRTH,
+        'citizen': _CITIZEN,
+        'lev_limit': _LEV_LIMIT,
+        'bmi': _BMI,
+        'smoking': _SMOKING,
+        'frequenc_tobacco': _FREQUENC_TOBACCO,
+        'frequenc_alcohol': _FREQUENC_ALCOHOL,
+        'n_portion': _N_PORTION,
+        'coicop': _COICOP,
+        'isced97': _ISCED97
+    }
+    return data_df.replace({column_name: mapper[column_name]})
```

Comment:
Can this be modified to replace all column values from the mapper list in a single call to replace_columns?

something like:
```
def replace_all_columns(df):
    df_columns = df.columns.to_list()
    mapper = {
       ...
    }
    for column, replacements in mapper.items():
       if column in df_columns:
            df.replace({column: replacements}, in_place=True)

   return df
```

Then the hardcoded list of columns repeated in EuroStat:_process_file can be removed.

Reply:
Made the changes.